java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -446,4 +449,83 @@ public CompletableFuture<String> functionLoadReplace(
                 route,
                 this::handleStringResponse);
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Object>[]> functionListWithCode(
+            @NonNull String libNamePattern) {
+        return commandManager.submitNewCommand(
+                FunctionList,
+                new String[] {LIBRARY_NAME_REDIS_API, libNamePattern, WITH_CODE_REDIS_API},
+                response -> castArray(handleArrayResponse(response), Map.class));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Object>[]> functionList(@NonNull String libNamePattern) {
+        return commandManager.submitNewCommand(
+                FunctionList,
+                new String[] {LIBRARY_NAME_REDIS_API, libNamePattern},
+                response -> castArray(handleArrayResponse(response), Map.class));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Object>[]> functionListWithCode() {
+        return commandManager.submitNewCommand(
+                FunctionList,
+                new String[] {WITH_CODE_REDIS_API},
+                response -> castArray(handleArrayResponse(response), Map.class));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Object>[]> functionList() {
+        return commandManager.submitNewCommand(
+                FunctionList,
+                new String[0],
+                response -> castArray(handleArrayResponse(response), Map.class));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Object>[]> functionListWithCode(
+            @NonNull String libNamePattern, @NonNull Route route) {
+        return commandManager.submitNewCommand(
+                FunctionList,
+                new String[] {LIBRARY_NAME_REDIS_API, libNamePattern, WITH_CODE_REDIS_API},
+                route,
+                response -> castArray(handleArrayResponse(response), Map.class));
```

Comment:
we don't need to consider single vs multi node responses here? 
Wouldn't we get a `ClusterValue` back?

Reply:
It was always single response. Maybe because it actually came from one node only and/or it was aggregated. Will re-test it.